pkg/alertmanager/operator.go

```diff
@@ -881,22 +882,32 @@ func (c *Operator) selectAlertManagerConfigs(ctx context.Context, am *monitoring
 					}
 				}
 
-				if pdConfig.HTTPConfig == nil {
-					continue
+				if pdConfig.HTTPConfig != nil {
+					if err = store.configureHTTPConfigInStore(ctx, pdConfig.HTTPConfig, amcNamespace, pdcKey); err != nil {
```

Comment:
I would update `configureHTTPConfigInStore()` to accept `nil` values for `*HTTPConfig` (and return early) as it would remove the need to test for `nil` everywhere.